sdk/src/quotes/public/swap-quote.ts

```diff
@@ -1,484 +1,185 @@
-    outputDelta = amountRemaining;
-  }
-
-  if (amountSpecified == AmountSpecified.Input && !hasReachedNextTick) {
-    inputDelta = amountRemaining;
-  } else {
-    inputDelta = inputDelta.add(calculateFeesFromAmount(inputDelta, feeRatePercentage));
-  }
-
-  return {
-    nextTickIndex,
-    nextSqrtPriceX64,
-    input: inputDelta,
-    output: outputDelta,
-    tickArraysCrossed: tickArraysCrossedUpdate,
-    hasReachedNextTick,
-  };
-}
-
-function calculateAmountAfterFees(amount: u64, feeRate: Percentage): BN {
-  return amount.mul(feeRate.denominator.sub(feeRate.numerator)).div(feeRate.denominator);
-}
-
-function calculateFeesFromAmount(amount: u64, feeRate: Percentage): BN {
-  return MathUtil.divRoundUp(
-    amount.mul(feeRate.numerator),
-    feeRate.denominator.sub(feeRate.numerator)
-  );
-}
-
-function calculateNewLiquidity(liquidity: BN, nextLiquidityNet: BN, swapDirection: SwapDirection) {
-  if (swapDirection == SwapDirection.AtoB) {
-    nextLiquidityNet = nextLiquidityNet.neg();
-  }
-
-  return liquidity.add(nextLiquidityNet);
-}
-
-function resolveTokenAmounts(
-  specifiedTokenAmount: BN,
-  otherTokenAmount: BN,
-  amountSpecified: AmountSpecified
-): [BN, BN] {
-  if (amountSpecified == AmountSpecified.Input) {
-    return [specifiedTokenAmount, otherTokenAmount];
-  } else {
-    return [otherTokenAmount, specifiedTokenAmount];
-  }
-}
-
-function fetchTickArray(baseInput: SwapSimulationBaseInput, tickIndex: number) {
-  const {
+    sqrtPriceLimit: SwapUtils.getDefaultSqrtPriceLimit(aToB),
+    otherAmountThreshold: SwapUtils.getDefaultOtherAmountThreshold(amountSpecifiedIsInput),
+    slippageTolerance,
     tickArrays,
-    whirlpoolData: { tickSpacing },
-  } = baseInput;
-
-  const startTickArray = tickArrays[0];
-  invariant(!!startTickArray, `tickArray is null at index 0`);
-  const sequenceStartIndex = startTickArray.startTickIndex;
-  const expectedArrayIndex = Math.abs(
-    Math.floor((tickIndex - sequenceStartIndex) / tickSpacing / TICK_ARRAY_SIZE)
-  );
-
-  invariant(
-    expectedArrayIndex > 0 || expectedArrayIndex < tickArrays.length,
-    `tickIndex ${tickIndex} assumes array-index of ${expectedArrayIndex} and is out of bounds for sequence`
-  );
-
-  const tickArray = tickArrays[expectedArrayIndex];
-  invariant(!!tickArray, `tickArray is null at array-index ${expectedArrayIndex}`);
-
-  return tickArray;
+  });
 }
 
-function fetchTick(baseInput: SwapSimulationBaseInput, tickIndex: number) {
-  const tickArray = fetchTickArray(baseInput, tickIndex);
-  const {
-    whirlpoolData: { tickSpacing },
-  } = baseInput;
-  return TickArrayUtil.getTickFromArray(tickArray, tickIndex, tickSpacing);
+/**
+ * Perform a sync swap quote based on the basic swap instruction parameters.
+ * @param params - SwapQuote parameters
+ * @returns a SwapQuote object with estimates on token amounts, fee & end whirlpool states.
+ */
+export function swapQuoteWithParams(params: SwapQuoteParam) {
```

Comment:
We need to maintain a sync-version of the quote function in case power users already have their own flow in fetching data prior to quoting. (ex. Jupiter)